README.md

```diff
@@ -365,6 +365,32 @@ Of course, the situation in which a class grows this long is strongly discourage
   }
   ```
 
+- If the only goal is to match on the type of the object, do NOT expand fully all the arguments, as it makes refactoring more difficult and the code more error prone.
```

Comment:
e.g., `WorkspaceId(workspaceId)` does not expect to be extended. Or `DbAwsKeys(accessKey, secretKey)`. They are fully self-contained concepts that would be as soon replaced as extended.